services/src/main/java/com/chatbot/services/ChatServiceController.java

```diff
@@ -0,0 +1,185 @@
+package com.chatbot.services;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.googleapis.auth.oauth2.GooglePublicKeysManager;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.chat.v1.model.ActionResponse;
+import com.google.api.services.chat.v1.model.Card;
+import com.google.api.services.chat.v1.model.Message;
+import com.google.api.services.chat.v1.model.Section;
+import com.google.api.services.chat.v1.model.TextParagraph;
+import com.google.api.services.chat.v1.model.WidgetMarkup;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.chatbot.services.protobuf.ChatServiceRequestOuterClass.ChatServiceRequest;
+import com.chatbot.services.protobuf.ChatServiceRequestOuterClass.ChatServiceRequest.ChatClient;
+import com.chatbot.services.protobuf.ChatServiceRequestOuterClass.ChatServiceRequest.MimeType;
+import com.chatbot.services.protobuf.ChatServiceRequestOuterClass.ChatServiceRequest.RequestType;
+
+@RestController
+public class ChatServiceController {
+
+  @Autowired
+  private AsyncService asyncService;
+
+  private static final String HANGOUTS_USER_AGENT = "Google-Dynamite";
+  private static final String REMOVED_FROM_SPACE_EVENT = "REMOVED_FROM_SPACE";
+  private static final String ADDED_TO_SPACE_EVENT = "ADDED_TO_SPACE";
+  private static final String MESSAGE_EVENT = "MESSAGE";
+  private static final String CARD_CLICKED_EVENT = "CARD_CLICKED";
+  private static final String CHAT_ISSUER = "chat@system.gserviceaccount.com";
```

Comment:
is this constant or a config? If config move it to something like properties and then refer.
same for one below the next one.

Reply:
updated